app/controllers/incomeSources/cease/IncomeSourceEndDateController.scala

```diff
@@ -262,33 +263,43 @@ class IncomeSourceEndDateController @Inject()(val authenticate: AuthenticationPr
           )(user, messages)))
         },
 
-        validatedInput => (incomeSourceType, id) match {
+        validatedInput =>
+          sessionService.createSession(JourneyType(Cease, incomeSourceType).toString).flatMap {
+            case true =>
+              (incomeSourceType, id) match {
+                case (SelfEmployment, None) =>
+                  val errorMessage: String = s"[IncomeSourceEndDateController][handleSubmitRequest]: missing income source ID - $id."
+                  Logger("application").error(s"${if (isAgent) "[Agent]"}" +
+                    s"$errorMessage")
+                  Future.failed(new Exception(s"$errorMessage"))
+                case (SelfEmployment, Some(incomeSourceId)) =>
+                  val result = Redirect(redirectAction)
+                  sessionService.setMongoKey(
+                    CeaseIncomeSourceData.dateCeasedField, validatedInput.date.toString, JourneyType(Cease, incomeSourceType)
+                  ).flatMap {
+                    case Right(true) => {
+                      sessionService.setMongoKey(
+                        CeaseIncomeSourceData.incomeSourceIdField, incomeSourceId, JourneyType(Cease, incomeSourceType)
+                      ).flatMap {
+                        case Right(true) =>
```

Comment:
this is also not reachable code (line 286), result is always true, so no need to test if this is true or false

Reply:
you can use Right(_) or any other approach, testing value if its "true" or "false" is not required and it a redundant code we should remove

Reply:
I don't think I'm testing if it's true or false anywhere? I'm flatmapping for the Left or Right return of sessionService.setMongoKey. I did Right(true) since that's the only return within the Right